### PR TITLE
runtime: it should rollback when failed in Sandbox AddInterface

### PR DESCRIPTION
**Pull request by yangfeiyu20102011**
When Sandbox AddInterface() is called, it may fail after endpoint.HotAttach,
we'd better rollback and call save() in the end.

Fixes: #3419 

Signed-off-by: yangfeiyu <yangfeiyu20102011@163.com>

### CODE REVIEW

**Comment by yangfeiyu20102011** [comment]
@liubin PTAL
Thanks.

File: src/runtime/virtcontainers/sandbox.go [hunk 1]
```diff
@@ -869,15 +869,28 @@ func (s *Sandbox) AddInterface(ctx context.Context, inf *pbTypes.Interface) (*pb
 		return nil, err
 	}
 
+	defer func() {
+		if err != nil {
+			if errDetach := endpoint.HotDetach(ctx, s.hypervisor, s.networkNS.NetNsCreated, s.networkNS.NetNsPath); errDetach != nil {
+				s.Logger().WithField("endpoint-type", endpoint.Type()).Error("rollback hot attaching endpoint failed")
```
Comment (jodh-intel) [inline src/runtime/virtcontainers/sandbox.go:875]: How about also logging the new error as that might give a clue about what went wrong:

```suggestion
				s.Logger().WithField("endpoint-type", endpoint.Type()).WithError(errDetach).Error("rollback hot attaching endpoint failed")
```

**Review by jodh-intel — APPROVED** [review]
Thanks for raising @yangfeiyu20102011 !

It would be great if you could also add a unit test for this change. However, there doesn't appear to be an existing test for this method and it looks like that could be rather awkward, so...

lgtm

File: src/runtime/virtcontainers/sandbox.go (hunk 1 above)
Comment (yangfeiyu20102011) [reply src/runtime/virtcontainers/sandbox.go]: OK，I have fixed it.

**Review by jodh-intel — APPROVED** [review]
Thanks @yangfeiyu20102011.

lgtm

**Review by devimc — APPROVED** [review]

**Pull request merged**